pkgs/tools/misc/diffoci/default.nix

```
@@ -16,12 +19,30 @@ buildGoModule rec {
 
   vendorHash = "sha256-18rsa91PiqZv70EK3K6K1l6N2mIpoVpkX29amKCo5cg=";
 
-  ldflags = [ "-s" "-w" ];
+  ldflags = [
+    "-s"
+    "-w"
+    "-X=github.com/reproducible-containers/${pname}/cmd/${pname}/version.Version=v${version}"
+  ];
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  postInstall =
+    let
+      diffoci = if stdenvNoCC.buildPlatform.canExecute stdenvNoCC.hostPlatform then "\"$out\"" else buildPackages.diffoci;
```

Most maintainers/contributors just let the shell completions error out when cross compiling.
But sure, this works as well.

Formatting nit (and `stdenvNoCC` replaced with `stdenv` because `buildGoModule` uses `stdenv` internally):

```suggestion
      diffoci = if stdenv.buildPlatform.canExecute stdenv.hostPlatform then placeholder "out" else buildPackages.diffoci;
```

Make sure to update it at the top of the file as well.

---

Done.